### ISSUE TITLE
Performance Drop Over Time / Intra-Epoch Checkpointing fails

### ISSUE THREAD

**Issue by dakami** [SeanNaren/deepspeech.pytorch#111]
So I'm training on the LibreSpeech dataset, warp-ctc disabled, dual Titan X Pascal.  It's taking about 1.2s/batch.  I'm checkpointing every 1000 batches.  Cancel, and reset:

Epoch: [7][5634/14063]	Time 0.409 (0.350)	Data 0.007 (0.011)	Loss 7.4160 (6.5093)	

There's a fairly hefty penalty to load the model, but once loaded it's quite a bit faster for at least an epoch.

Could be memory fragmentation, could be resuming mid-epoch isn't entirely working.  If it is working, you could probably script the restart and 2x-3x perf.

Here is my full invocation (I do have 96 HT cores).

python train.py --train_manifest data/libri_train_manifest.csv --val_manifest data/libri_test_manifest.csv  --cuda  --checkpoint  --visdom --num_workers 96 --rnn_type gru --hidden_size 800 --hidden_layers 5  --learning_anneal 1.01 --epochs 100 --checkpoint_per_batch 1000 --continue_from models/deepspeech_checkpoint_epoch_7_iter_6000.pth.tar 

Thank you!  This is so much fun to play with!

**Comment by dakami** [comment]
It does speed up between epochs though.  I know manifests are sorted by time, maybe resuming mid-epoch doesn't start loading the appropriate samples?

**Comment by SeanNaren** [comment]
I am unsure as to what exactly is the issue here, could you further explain?

**Comment by SeanNaren** [comment]
Going to close this for now...
